packages/webapp/src/components/Animals/AddAnimalsFormCard/AddAnimalsFormCard.tsx

```diff
@@ -112,6 +114,7 @@ export default function AddAnimalsFormCard({
         <NumberInput
           name={`${namePrefix}${BasicsFields.COUNT}`}
           control={control}
+          max={ANIMAL_COUNT_LIMIT}
```

Comment:
Thank you for working on this!

![Screenshot 2024-11-20 at 3 55 23 PM](https://github.com/user-attachments/assets/9171f50a-33c3-4ac6-9394-aefc7516d9cf)

Sorry, we should've explained this first... When the checkbox is selected and the count is 1000, 1000 animals will be created. If the checkbox is not selected, a batch with 1000 animals (one record in the DB) will be created instead. The latter is a valid case, so it would be great if you could add a condition to set a maximum limit when the checkbox is selected 🙏

Reply:
Hi, Sayaka. So I put a hook to deal with this, but then I found it wouldn't change the displayed value in the input when I only use setValue, (i.e.  if number is 2000 and user clicks the checkbox, the value will be set to the limit, but still displayed as 2000) so I put a "value" prop for the NumberedInput, let me know what you think about it.

Reply:
Sorry for the late reply! Sending `value` to `NumberInput` seems good, but it looks like `value` will be automatically passed through without modifying `NumberInput`, since `...props` is added to `InputBase`. Could you double check?

Would it work if we move the logic from the `useEffect` to the checkbox's `onChange` instead?

Reply:
Oh right, `inputProps`'s value overrides it.

@navDhammu The number in the `NumberInput` doesn't update when the value is changed via `setValue`. Do you think we should update the logic in `useNumberInput` to handle this?

Reply:
@SayakaOno The reason it doesn't update with `setValue` is because `NumberInput` manages its own state, so when changing value with setValue it doesn't update NumberInput's internal state, which is why you still see the previous number. 

I don't think its a good idea to update the logic inside the `useNumberInput` to handle this because then you would probably have to add a useEffect to update the state based on changes to some prop, and this isn't best practices for react. 

Better ways of handling this in my opinion:
1. Lift the state up to where its needed. This would involve refactoring out the useNumberInput hook from NumberInput and using it in each component that uses NumberInput.
2. Use a key to rerender NumberInput.

Also, sending `value` to `NumberInput` is not a good idea because it overrides the value given by `useNumberInput` which is a formatted string based on the locale. So overriding that breaks the localization feature.

Reply:
Thank you for your input, I'll discuss the solution with the team!

Reply:
@navDhammu Thanks for explaining.  @SayakaOno I think the 2nd way of using a key is the easiest.

Reply:
I agree, but I'm hesitant to go with that approach because we may encounter the same issue in the future. Needing to add a key each time doesn't seem like a sustainable solution...

I'll bring this up in tech daily early next week (feel free to join if you'd like) and get back to you!

Reply:
A bit late to the party but chiming in here! I actually think changing the value of the input ourselves wouldn't be a good UX practice, because this happens inadvertently for the user and they get no context on why the number is changing and no visibility that there's a limit on the field. Instead, we should validate if the user's input is higher than the limit and if it is, show the input as errored and show an error message. Per [this comment](https://lite-farm.atlassian.net/browse/LF-4410?focusedCommentId=17718) the message should be “You can only add up to 1000 animals at a time”.